packages/jest-runtime/src/transform.js

```diff
@@ -276,7 +246,12 @@ const cachedTransformAndWrap = (
   if (preprocessor && shouldPreprocess(filename, config)) {
     result = preprocessor.process(result, filename, config);
   }
-  if (shouldInstrument(filename, config)) {
+
+  // That means that the preprocessor has a custom instrumentation
+  // logic and will handle it based on `config.collectCoverage` option
+  const preprocessorWillInstrument = preprocessor && preprocessor.INSTRUMENTS;
+
+  if (!preprocessorWillInstrument && shouldBeCovered(filename, config)) {
```

Comment:
i thought that caching would not worth it, because it can eat a lot of memory with little benefit (# of .js files times # of workers)
but let me try